janus_messages/src/lib.rs

```diff
@@ -419,15 +424,17 @@ impl Decode for Nonce {
 
 /// Checksum over DAP report nonces, defined in §4.4.4.3.
 #[derive(Copy, Clone, Debug, Default, Hash, PartialEq, Eq)]
-pub struct NonceChecksum([u8; SHA256_OUTPUT_LEN]);
+pub struct NonceChecksum([u8; 32]);
```

Comment:
`NonceChecksum` does appear in the protocol message `AggregateShareReq`. However we could make `AggregateShareReq.checksum` an opaque `[u8; 32]` to keep the `NonceChecksum` type and its logic out of `janus_messages`.

Reply:
So, the DAP spec doesn't define a type called `NonceChecksum` at all -- `AggregateShareReq.checksum` is defined as `opaque checksum[32]`. We made the implementation choice to define our own type because (I suppose) that makes it easy to attach behavior to the type, but it's probably not appropriate to push the type & behavior onto folks in a message-serialization library.

That said, this is a pretty pedantic distinction -- especially now that the functionality has been moved out. If you don't find my argument moving, or if it's just more work than it's worth, I won't block exporting this type. :)